sites/docs/docs/features/queries/query-chaining.md

```diff
@@ -17,33 +17,33 @@ For example, if you want to reference a query named `daily_sales`, you would wri
 Here we have a normal SQL query, called *sales_by_region:*
 
 ```sql
-select 
+select
     region,
     sum(sales) as sales
-from production.daily_sales 
+from production.daily_sales
 group by 1
 ```
 
 Next, we have a query calculating average sales across regions, which references *sales_by_region* using the `${ }` syntax.
 
 ```sql
-select 
+select
     avg(sales) as average_sales
 from ${sales_by_region}
 ```
 
 Below is the compiled SQL that's sent to the database:
 
 ```sql
-select 
+select
     avg(sales) as average_sales
 from (
-    select 
+    select
         region,
         sum(sales) as sales
-    from production.daily_sales 
-    group by 1 
-)
+    from production.daily_sales
+    group by 1
+) as t
```

Comment:
Agree that this would not run in many dialects. 
However, our compiler does not **_currently_** add the alias if the user does not (maybe it should). 

If you try this in Postgres for example, it will throw the error `Subquery in FROM must have an alias`

For clarity in the docs though, I think if we are going to add this alias, we need to add it to the query above also!

ie `from ${sales_by_region} as t`

Reply:
I'm projecting my dbt experience on this, so I'm expecting it to work the same way (`ephemeral` materialization).

If you're not using an alias, the compiler generates one for you:

```sql
select a, b, c
from ${some_table}
---
select a, b, c
from (select * from some_table) as generated_alias
```

And if you're using one, the compiler will use that:

```sql
select t.a, t.b, t.c
from ${some_table) as t
---
select t.a, t.b, t.c
from (select * from some_table) as t
```

So ideally it needs to work both ways.

Reply:
Okay great - I'm going to add an issue to track this

Reply:
Issue added [here](https://github.com/evidence-dev/evidence/issues/427). For now I'm just going to include a note in the docs about a sub-query alias being needed in some dialects.